modular_zzplurt/readme.md

```diff
@@ -0,0 +1,165 @@
+2. **Use commenting conventions for non-modular edits:** When editing non-modular code, make sure to adequately use the commenting conventions to ensure it is known that it's a non-modular edit.
+3. **It's ok to mess up:** These rules are intended less as a set of guidelines for contributors to strictly follow, but rather coding standards that we all collectively help to attain, for examble through maintainers helping any contributors fulfill them in their pull requests.
+
+### Modular Overrides
+
+Modular overrides allow us to extend or modify the behavior of existing code without directly editing the core files. This is done by creating new definitions or modifying existing ones in a way that they can be easily integrated into the core codebase. There are two main types of modular overrides: Variable Overrides and Proc Overrides.
+
+#### Variable Overrides
+
+Variable overrides allow you to modify existing variables in a modular fashion. This is useful when you need to introduce new properties to existing objects without altering the core code. For example:
+
+```byond
+/obj/item/gun
+    var/muzzle_flash = TRUE
+
+    ...
+
+/obj/item/gun
+    muzzle_flash = FALSE
+```
+
+In this example, a new variable muzzle_flash is added to the /obj/item/gun object with a value of `TRUE`. The value of this variable is later changed to `FALSE` through overriding.
+
+#### Proc Overrides
+
+Proc overrides allow you to extend or modify the behavior of existing procedures (procs) without directly editing the core files. This is done by defining a new proc that calls the original proc and then adds the new behavior. For example:
+
+```byond
+/obj/item/gun/shoot_live_shot(mob/living/user, pointblank = 0, atom/pbtarget = null, message = 1)
+    . = ..() // Call the original proc and set its value to the default return value
+    if(muzzle_flash)
+        spawn_sparks(src) // Add new behavior
+```
+
+In this example, the shoot_live_shot proc is overridden to add a new behavior (spawning sparks) after calling the original proc. This ensures that the new behavior is added without modifying the core code.
+
+#### When to Use Modular Overrides
+
+Modular overrides should be used whenever possible to keep the core codebase clean and maintainable. However, it's important to use them judiciously and ensure that they do not introduce performance issues or make the code harder to understand and maintain. Specifically:
+
+- **Use modular overrides when they won't cause more issues than not modularizing them:** If a modular override would introduce significant complexity, performance issues, or maintenance challenges, it may be better to make a non-modular edit with proper commenting.
+- **Avoid unnecessary overrides:** Do not override procs or variables unless it is necessary for your feature or fix. Unnecessary overrides can lead to maintenance headaches and potential conflicts with upstream changes.
```

Comment:
Is `Do not override procs or variables unless it is necessary for your feature or fix.` really worth saying? I feel like everyone adding something thinks it's necessary; it'd be better to say something like
```suggestion
- **Avoid unnecessary overrides:** Do not override procs or variables unless it is the only reasonable way to implement your feature or fix. Unnecessary overrides can lead to maintenance headaches and potential conflicts with upstream changes.
```
That ties in nicely with the first pillar in `## The Modularization Protocol` above, in that modular edits and overrides should only be used when they're the most reasonable option. We could even add a link to that section here (or have an entirely new section explaining what it means for a change to be reasonable, if necessary).

Reply:
Yeah, the idea that the modularization protocols should convey to coders in that regard is "Always try first modularizing your changes. Only edit non modular code if it's not possible to modularize or if modularizing would worsen the code compared to not modularizing."